### PR TITLE
documenting property and analyser

### PR DESCRIPTION
**Pull request by yogeshg**

### CODE REVIEW

File: include/property.h [hunk 1]
```diff
@@ -19,74 +22,187 @@ template <typename T> concept bool StringRef = requires(T o) {
...
+  CodeContext(const std::string &a, const std::string &b, const std::string &c,
+            const int &d, const std::string &e)
+      : file(a), func(b), level(c), line(d), hash(e) {}
+
+  /**
+   * @brief Returns a tuple of all the members
+   * @details This returns a serialisable representation that can be used to
+   * print or order this class
+   * 
+   * @returns a tuple with all the constituents
```
Comment (andylhxu) [inline include/property.h:78]: This should be @return

**Review by andylhxu — CHANGES_REQUESTED** [review]

**Review by andylhxu — APPROVED** [review]

File: include/property.h (hunk 1 above)
Comment (andylhxu) [reply include/property.h:78]: actually returns will also work

Comment (yogeshg) [reply include/property.h:78]: oh did not know either facts

**Comment by yogeshg** [comment]
@andylhxu more docs

**Pull request merged**